src/ol_infrastructure/components/aws/auto_scale_group.py

```diff
@@ -0,0 +1,345 @@
+from typing import Dict, List, Optional, Union
+
+import pulumi
+from pulumi_aws.acm import get_certificate
+from pulumi_aws.autoscaling import (
+    Group,
+    GroupInstanceRefreshArgs,
+    GroupInstanceRefreshPreferencesArgs,
+    GroupLaunchTemplateArgs,
+    GroupTagArgs,
+)
+from pulumi_aws.ec2 import (
+    LaunchTemplate,
+    LaunchTemplateBlockDeviceMappingArgs,
+    LaunchTemplateBlockDeviceMappingEbsArgs,
+    LaunchTemplateIamInstanceProfileArgs,
+    LaunchTemplateTagSpecificationArgs,
+    SecurityGroup,
+)
+from pulumi_aws.lb import (
+    Listener,
+    ListenerArgs,
+    ListenerDefaultActionArgs,
+    LoadBalancer,
+    TargetGroup,
+    TargetGroupHealthCheckArgs,
+)
+from pydantic import BaseModel, NonNegativeInt, PositiveInt, validator
+
+from bridge.lib.magic_numbers import (
+    AWS_LOAD_BALANCER_NAME_MAX_LENGTH,
+    AWS_TARGET_GROUP_NAME_MAX_LENGTH,
+    DEFAULT_HTTPS_PORT,
+)
+from ol_infrastructure.lib.aws.ec2_helper import DiskTypes, InstanceTypes
+from ol_infrastructure.lib.ol_types import AWSBase
+
+
+class BlockDeviceMapping(BaseModel):
+    """Container for describing a block device mapping for an EC2 instance/launch template"""
+
+    delete_on_termination: bool = True
+    device_name: str = "/dev/xvda"
+    volume_size: PositiveInt = PositiveInt(25)
+    volume_type: DiskTypes = DiskTypes.legacy_ssd
```

Comment:
Why are we defaulting to the gp2 disk type?

Reply:
That is a good question to which I do not have an answer. Is gp3 more appropriate for a default?

Reply:
Yeah, we should default to gp3 because it's generally cheaper with better performance than gp2. It's the newest generation of generic SSD so we should bias toward that unless there's a reason that we can't.